### PR TITLE
Self-review of the recent Trainer changes

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Follow-up to #14862 and #14871

### Does your PR introduce any breaking changes? If yes, please list them.

None, as this is unreleased code.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @justusschock @kaushikb11 @awaelchli @borda @ananthsub @ninginthecloud @jjenniferdai @rohitgr7 @akihironitta

### CODE REVIEW

File: src/pytorch_lightning/trainer/call_fn.py [hunk 1]
```diff
@@ -14,13 +14,14 @@
 import traceback
```
Comment (carmocca) [inline src/pytorch_lightning/trainer/call_fn.py:14]: Open to suggestions for a different name, but `teardown` is not the main responsibility of this function

**Review by akihironitta — APPROVED** [review]

File: src/pytorch_lightning/trainer/call_fn.py (hunk 1 above)
Comment (awaelchli) [reply src/pytorch_lightning/trainer/call_fn.py:14]: I would not rename the file if Will specifically called it this way.

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/pytorch_lightning/trainer/call.py [hunk 2]
```diff
@@ -14,13 +14,14 @@
...
 
+import pytorch_lightning as pl
 from lightning_lite.utilities.distributed import distributed_available
 from pytorch_lightning.trainer.states import TrainerStatus
 from pytorch_lightning.utilities.exceptions import _TunerExitException
 from pytorch_lightning.utilities.rank_zero import rank_zero_warn
 
 
-def call_and_handle_interrupt(trainer: Any, trainer_fn: Callable, *args: Any, **kwargs: Any) -> Any:
+def _call_and_handle_interrupt(trainer: "pl.Trainer", trainer_fn: Callable, *args: Any, **kwargs: Any) -> Any:
```
Comment (Borda) [inline src/pytorch_lightning/trainer/call.py:24]: API chnage?

File: src/pytorch_lightning/trainer/trainer.py [hunk 3]
```diff
@@ -66,7 +66,7 @@
 )
 from pytorch_lightning.profilers import Profiler
 from pytorch_lightning.strategies import ParallelStrategy, Strategy
-from pytorch_lightning.trainer import setup, teardown
+from pytorch_lightning.trainer import call, setup
```
Comment (Borda) [inline src/pytorch_lightning/trainer/trainer.py:69]: This could be a breaking change for some advanced users...

**Pull request merged**